### PR TITLE
Add (U.S.A.) State JSX component

### PR DESCRIPTION
**Pull request by andygout**
### Description
This PR creates an **(U.S.A.) State** JSX component that matches the behaviour and usage of the existing Handlebars (HBS) partial, with tests to ensure they match.

[Trello ticket](https://trello.com/c/WYujLFFR/1597-n-conversion-forms-new-major).

### Reminder
Have you completed these common tasks (remove those that don't apply)?

- [ ] **Documentation** updated or created
- [x] **Tests** written for new or updated for existing functionality
- [x] **Demos** updated to use this change
- [ ] **Accessibility** checked for screen readers and contrast
- [ ] **Design Review** ran past the designer 
- [ ] **Product Review** ran past the product owner

### CODE REVIEW

File: components/index.jsx [hunk 1]
```diff
@@ -53,5 +54,7 @@ export {
 	Position,
 	RegistrationConfirmation,
 	Submit,
-	TrialBanner
+	TrialBanner,
+	State,
+	Submit
```
Comment (umbobabo) [inline components/index.jsx]: This line need to be removed cause Submit is already there.

File: components/state.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+State.PropTypes = {
+	value: PropTypes.string,
+	fieldId: PropTypes.string,
+	selectId: PropTypes.string,
+	hasError: PropTypes.bool,
+	isHidden: PropTypes.bool,
+	isBillingState: PropTypes.bool,
+	isDisabled: PropTypes.bool,
+	states: PropTypes.arrayOf(PropTypes.shape({
+		code: PropTypes.string,
```
Comment (umbobabo) [inline components/state.jsx]: Good stuff 👍

File: components/index.jsx (hunk 1 above)
Comment (andygout) [reply components/index.jsx]: Rebase went wrong. Have fixed.

File: partials/state.html [hunk 3]
```diff
@@ -13,9 +13,9 @@
 		data-trackable="field-state"
 		{{#if isDisabled}}disabled{{/if}}>
 
-		<option disabled selected value>Please select a state</option>
+		<option disabled value="">Please select a state</option>
```
Comment (andygout) [inline partials/state.html]: @umbobabo Is it okay to have removed `selected` here? (It was causing me all sorts of problems when diff-ing the HBS and JSX outputs.)

`option` elements in similar components (e.g. `industry.jsx`) do not have this attribute and neither does it make sense to have the attribute when the element also has a `disabled` attribute.

**Pull request merged**